test/py/test_runner.py

```diff
@@ -0,0 +1,192 @@
+
+    return args
+
+
+def get_sub_folders(dir_name):
+    dir_contents = os.listdir(dir_name)
+    folders = []
+    for item in dir_contents:
+        tmp_item = dir_name + '/' + item
+        if os.path.isdir(tmp_item):
+            folders.append(item)
+    folders.sort()
+
+    return folders
+
+
+def get_test_cases(dir_name):
+    return get_sub_folders(dir_name)
+
+
+def get_model_name(dir_name):
+    dir_contents = os.listdir(dir_name)
+    for item in dir_contents:
+        file_name = dir_name + '/' + item
+        if os.path.isfile(file_name) and file_name.endswith('.onnx'):
+            return item
+
+    return ''
+
+
+def read_pb_file(filename):
+    try:
+        pfile = open(filename, 'rb')
+    except IOError:
+        print("File {} open error".format(filename))
+        sys.exit(1)
+
+    data_str = pfile.read()
+    tensor = onnx.TensorProto()
+    tensor.ParseFromString(data_str)
+    np_array = numpy_helper.to_array(tensor)
+
+    return np_array
+
+
+def wrapup_inputs(io_folder, parameter_names):
+    index = 0
+    param_map = {}
+    for param_name in parameter_names:
+        file_name = io_folder + '/input_' + str(index) + '.pb'
+        data = read_pb_file(file_name)
+        param_map[param_name] = data
+        index = index + 1
+
+    return param_map
+
+
+def read_outputs(io_folder, out_num):
+    outputs = []
+    for i in range(out_num):
+        file_name = io_folder + '/output_' + str(i) + '.pb'
+        data = read_pb_file(file_name)
+        outputs.append(data)
+
+    return outputs
+
+
+def run_one_case(model, param_map):
+    # convert np array to model argument
+    pp = {}
+    for key, val in param_map.items():
+        print("input = {}".format(val))
+        pp[key] = migraphx.argument(val)
+
+    # run the model
+    model_outputs = model.run(param_map)
+
+    # convert argument to np array
+    outputs = []
+    for output in model_outputs:
+        outputs.append(np.array(output))
+
+    return outputs
+
+
+def check_correctness(gold_outputs, outputs, rtol=1e-3, atol=1e-3):
+    if len(gold_outputs) != len(outputs):
+        print("Number of outputs {} is not equal to expected number {}".format(
+            len(outputs), len(gold_outputs)))
+        return False
+
+    out_num = len(gold_outputs)
+    ret = True
+    for i in range(out_num):
+        print("Expected value: \n{}".format(gold_outputs[i]))
+        print("Actual value: \n{}".format(outputs[i]))
+        if not np.allclose(gold_outputs[i], outputs[i], rtol, atol):
+            print("Output {} is incorrect ...".format(i))
+            print("Expected value: \n{}".format(gold_outputs[i]))
+            print("Actual value: \n{}".format(outputs[i]))
+            ret = False
+
+    return ret
+
+
+def tune_input_shape(model, input_data):
+    input_data_keys = input_data.keys()
+    param_shapes = model.get_parameter_shapes()
+    input_shapes = {}
+    for name, s in param_shapes.items():
+        assert name in input_data_keys
```

Comment:
I think we can fix this deepcode warning and just say `assert name in input_data`

Reply:
fixed